utils/commands.py

```diff
@@ -2,12 +2,15 @@
 from discord.ext import commands
 
 async def available_subcommands(ctx: commands.Context):
+    """
+    Send a list of available subcommands in a group command.
+    """
     subcmds = []
     for cmd in ctx.command.commands:
         try:
             usable = await cmd.can_run(ctx)
             if usable:
                 subcmds.append('`'+cmd.name+'`')
-        except commands.CommandError as e:
-            logging.debug(f'Command "{cmd.name}" check threw error, discarded in {ctx.command.name} group subcommand list.')
+        except commands.CommandError:
+            logging.exception(f'Command "{cmd.name}" check threw error, discarded in {ctx.command.name} group subcommand list.')
```

Comment:
> It might be valuable to log the error thrown as well here.

In hindsight, I wasn't the clearest here - I meant including the exception in the logged message, something like this:
```py
except commands.CommandError as e:
    logging.debug(f'Command "{cmd.name}" check threw error, discarded in {ctx.command.name} group subcommand list.\n\t{e}')
```
I think checks throwing exceptions is to be expected here so it's best to log at a `DEBUG` level, but logging the specific exception thrown could trace which check threw the error.

Reply:
What do you mean by "checks throwing exceptions is to be expected here"?

Reply:
Like, when I was testing the updates cog, I remember discord.py's built-in permission checks throwing exceptions to signal a check failure.

Reply:
Hm so is the functionality provided by `logging.exception` not desired then? It prints the exception/stack trace as well [below specified text](https://stackoverflow.com/questions/5191830/how-do-i-log-a-python-error-with-debug-information).

Reply:
Ah, I get it - that is a much better solution than the one in my comment. However, `logging.exception` logs at an `ERROR` level so it is still not quite the desired functionality - using `exc_info=True` inside `logging.debug` seems like a good idea (else this will log at an `ERROR` level every time a user lists subcommands and does not have permissions for all of them).

Reply:
Right good catch, should be changed now.